lib/pbench/agent/utils.py

```diff
@@ -1,41 +1,60 @@
+import logging
 import os
-import sys
-import pathlib
 
-import six
+import colorlog
 
-from pbench.agent.logger import logger
-
-tools_group_prefix = "tools-v1"
 
+def setup_logging(name=None, debug=False, logfile=None):
+    """Setup logging for client
+    :param None: name of the python object
+    :param debug: Turn on debug logging
+    :param logfile: Logfile to write to
+    """
+    if not name:
+        log = logging.getLogger()  # root logger
+    else:
+        log = logging.getLogger(name)
 
-def init_wrapper(config):
-    """Initialize agent envrionment before running a command
+    # Make sh logging a bit less verbose
+    logging.getLogger("sh").setLevel(logging.WARNING)
 
-    :param config: a configparser object
-    """
-    if six.PY2:
-        logger.error("python3 is required, either directly or through SCL")
-        sys.exit(1)
-
-    pbench_run = pathlib.Path(config.rundir)
-    if pbench_run.exists():
-        # Its possible to run pbench without root
-        # but check to make sure that the rundir is writable
-        # before we do anything else
-        if os.access(pbench_run, os.W_OK) is not True:
-            logger.error("%s is not writable", pbench_run)
-            sys.exit(1)
-        pbench_tmp = pathlib.Path(pbench_run, "tmp")
-        if not pbench_run.exists():
-            # the pbench temporary directory is always relative to pbench run
-            pbench_tmp.mkdir(parents=True, exists_ok=True)
+    if (
+        os.environ.get("_PBENCH_TOOL_MEISTER_CLIENT_LOG_LEVEL") == "debug"
+        or os.environ.get("_PBENCH_TOOL_MEISTER_START_LOG_LEVEL") == "debug"
+        or os.environ.get("_PBENCH_TOOL_MEISTER_START_LOG_LEVEL") == "debug"
```

Comment:
Same comments as before.

1. Although I suspect that fine tuning log levels is rarely all that useful, you're overriding that option by bundling them together.
2. You're checking `_PBENCH_TOOL_MEISTER_START_LOG_LEVEL` twice.

Reply:
Fixed